modules/nf-core/vgan/haplocart/main.nf

```diff
@@ -0,0 +1,43 @@
+
+process VGAN_HAPLOCART {
+    tag "$meta.id"
+    label 'process_low'
+
+    conda (params.enable_conda ? "bioconda::vgan=1.0.1" : null)
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/vgan:1.0.1--h9ee0642_0':
+        'quay.io/biocontainers/vgan:1.0.1--h9ee0642_0' }"
+
+    input:
+    tuple val(meta), path(hc_files), path(reads), path(reads2)
+
+    output:
+    tuple val(meta), path("*.txt"), path("*.posterior.txt"), emit: txt
+    path "versions.yml"           , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    def VERSION = '1.0.1'
+    def reads2_path = reads2 ?: reads
+    def reads_args = meta.single_end ? "-fq1 ${reads}" : "-fq1 ${reads} -fq2 ${reads2_path}"
```

Comment:
```suggestion
    def reads_args = meta.single_end ? "-fq1 ${reads}" : "-fq1 ${reads} -fq2 ${reads2}"
```
No need in making it so complicated. People should set their meta fields right

Reply:
Indeed (note @JoshuaDanielRubin is new to nf-core ;) )

What this means is that your Nextflow input channel object e.g. `${reads}` can either be a single file _or_ a list. In nf-core tooling, we track whether there is single or paired-end files via a special field called `meta.single_end`, so if single_end is set to true, only a single file is expected so just need  a single `-fq1`), but if it's set to false then it's expected it is paired-end :)

Reply:
Yeah my bad I should have put some more explanation on my comments (guess this was the last thing I did before my off day, so probably rushed it a bit :grin:)